code/modules/clothing/under/accessories/storage.dm

```diff
@@ -46,6 +56,10 @@
 	..()
 	loc = master_item
 
+/obj/item/weapon/storage/internal/Destroy()
+	master_item = null
+	return ..()
+
 /obj/item/clothing/accessory/storage/webbing
 	name = "webbing"
 	desc = "Strudy mess of synthcotton belts and buckles, ready to share your burden."
```

Comment:
Ah. Yes, that would be a problem.
I'll just go ahead and give the base storage accessory a `cant_hold` list that includes storage accessories and anything they can be attached to.